### DISCUSSION TITLE
Custom index.html

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#4698] [Q&A] [answered]
Hi there,

Does the framework support custom index.html and allow ejs like syntax in it? A use case would be to include certain html header element based on the configuration value or environment variable. Would be to know if we are able to make some variables available to read from within index.html. Below is an example of Vite use case:

Ejs condition in index.html (Read variable injected in vite.config.ts)
![image](https://github.com/web-infra-dev/modern.js/assets/52765572/a121d58d-3acf-425e-a0b4-64b2d4edd19f)

Vite config provide variables
![image](https://github.com/web-infra-dev/modern.js/assets/52765572/8a01ccb7-e7f2-4b6a-8d7c-ec0a7bf6abc0)

thanks! :)

**Comment by chenjiahan** [comment]
Modern.js supports EJS syntax by default, you can refer to this document: https://modernjs.dev/en/guides/basic-features/html.html#html-syntax

**Reply by AsuraKev** [reply]
@chenjiahan thanks! the link you shared isnt working. :(

Do you know where the index.html is when we generate a project with Modern.js Cli?

**Reply by chenjiahan** [reply]
Sorry I shared a bad link...the correct link is https://modernjs.dev/en/guides/basic-features/html.html#html-syntax

**Reply by chenjiahan** [reply]
You can create an `/config/html/index.html` file and it will replace the default HTML template.

**Reply by AsuraKev** [reply]
Awesome @chenjiahan. How about create a /src/document.tsx? Will that override default index.html

**Reply by chenjiahan** [reply, marked as answer]
Of course, `/src/Document.tsx` will override the default `index.html` too.

**Reply by AsuraKev** [reply]
thanks for the quick response!
